src/jvm/main/org/jetbrains/kotlinx/lincheck/util/AnalysisSections.kt

```diff
@@ -182,23 +184,157 @@ internal inline fun <R> runOutsideIgnoredSection(block: () -> R): R {
+    className.startsWith("kotlin.") -> AnalysisSectionType.IGNORED
+
+    // All other classes should be transformed
+    else -> AnalysisSectionType.NORMAL
 }
 
+internal fun isCollectionsLibrary(className: String) = className in setOf(
+    // Interfaces
+    "java.lang.Iterable",
+    "java.util.Collection",
+    "java.util.List",
+    "java.util.Set",
+    "java.util.Queue",
+    "java.util.Deque",
+    "java.util.NavigableSet",
+    "java.util.SortedSet",
+    "java.util.Map",
+    "java.util.SortedMap",
+    "java.util.NavigableMap",
+    
+
+    // Abstract implementations
+    "java.util.AbstractCollection",
+    "java.util.AbstractList",
+    "java.util.AbstractQueue",
+    "java.util.AbstractSequentialList",
+    "java.util.AbstractSet",
+    "java.util.AbstractMap",
+
+    // Concrete implementations
+    "java.util.ArrayDeque",
+    "java.util.ArrayList",
+    "java.util.AttributeList",
+    "java.util.EnumSet",
+    "java.util.HashSet",
+    "java.util.LinkedHashSet",
+    "java.util.LinkedList",
+    "java.util.PriorityQueue",
+    "java.util.Stack",
+    "java.util.TreeSet",
+    "java.util.Vector",
+    "java.util.HashMap",
+    "java.util.LinkedHashMap",
+    "java.util.WeakHashMap",
+    "java.util.TreeMap",
+)
+
+internal fun isConcurrentCollectionsLibrary(className: String) = className in setOf(
+    // Interfaces
+    "java.util.concurrent.BlockingDeque",
+    "java.util.concurrent.BlockingQueue",
+    "java.util.concurrent.TransferQueue",
+    "java.util.concurrent.ConcurrentMap",
+    "java.util.concurrent.ConcurrentNavigableMap",
+
+    // Concrete implementations
+    // Concurrent collections
+    "java.util.concurrent.ConcurrentHashMap",
+    "java.util.concurrent.ConcurrentLinkedDeque",
+    "java.util.concurrent.ConcurrentLinkedQueue",
+    "java.util.concurrent.ConcurrentSkipListMap",
+    "java.util.concurrent.ConcurrentSkipListSet",
+
+    // Blocking queues
+    "java.util.concurrent.ArrayBlockingQueue",
+    "java.util.concurrent.LinkedBlockingDeque",
+    "java.util.concurrent.LinkedBlockingQueue",
+    "java.util.concurrent.DelayQueue",
+    "java.util.concurrent.PriorityBlockingQueue",
+    "java.util.concurrent.SynchronousQueue",
+    "java.util.concurrent.LinkedTransferQueue",
+
+    // Copy-on-write collections
+    "java.util.concurrent.CopyOnWriteArrayList",
+    "java.util.concurrent.CopyOnWriteArraySet",
+
+    // Inner class view
+    "java.util.concurrent.ConcurrentHashMap\$KeySetView"
+)
+
 private fun isJavaExecutorService(className: String) =
-    className.startsWith("java.util.concurrent.AbstractExecutorService") ||
-    className.startsWith("java.util.concurrent.ThreadPoolExecutor") ||
-    className.startsWith("java.util.concurrent.ForkJoinPool")
\ No newline at end of file
+    className.startsWith("java.util.concurrent.AbstractExecutorService") 
+        || className.startsWith("java.util.concurrent.ThreadPoolExecutor") 
+        || className.startsWith("java.util.concurrent.ForkJoinPool")
```

Comment:
Is this formatting change necessary?
IMO, it looks better when all lines have same indentation.

Reply:
The `||` used to be at the end of previous lines which is confusing IMO. And IJ decided to use this indentation level. But do agree that this extra tab is not ideal.